examples/gno.land/r/gnoland/coins/coins.gno

```diff
@@ -0,0 +1,112 @@
+// Package coins provides simple helpers to retrieve information about coins
+// on the Gno.land blockchain.
+//
+// The primary goal of this realm is to allow users to check their token balances without
+// relying on external tools or services. This is particularly valuable for new networks
+// that aren't yet widely supported by public explorers or wallets. By using this realm,
+// users can always access their balance information directly through the gnodev.
+//
+// While currently focused on basic balance checking functionality, this realm could
+// potentially be extended to support other banker-related workflows in the future.
+// However, we aim to keep it minimal and focused on its core purpose.
+//
+// This is a "Render-only realm" - it exposes only a Render function as its public
+// interface and doesn't maintain any state of its own. This pattern allows for
+// simple, stateless information retrieval directly through the blockchain's
+// rendering capabilities.
+//
+// Example usage:
+//
+//	/r/gnoland/coins:ugnot - shows the total supply of ugnot
+//	/r/gnoland/coins:ugnot/g1... - shows the ugnot balance of a specific address
+package coins
+
+import (
+	"std"
+	"strings"
+)
+
+// TotalSupply returns the total supply of the specified denomination
+func TotalSupply(banker std.Banker, denom string) int64 {
```

Comment:
can we have this API instead, and always use a readonly banker?
```suggestion
func TotalSupply(denom string) int64 {
```

Reply:
Modified it so that the banker object is created inside the function. I also considered placing the banker globally, but it seemed better to restrict the object's lifetime to within the function.
https://github.com/gnolang/gno/pull/3899/commits/695412e8145fe483eff08744fd8a190578a4230c